### PR TITLE
Update the market data streams

### LINKED ISSUE #309: Update StockDataStream to support statuses
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

Currently, [`StockDataStream`](https://alpaca.markets/docs/python-sdk/api_reference/data/stock/live.html#stockdatastream) doesn't look to provide [statuses](https://alpaca.markets/docs/api-references/market-data-api/stock-pricing-data/realtime/#trading-status). Without this, it is not possible to know of trading halts.

### Describe the solution you'd like.

Ideally, there could perhaps be `subscribe_statuses` and `unsubscribe_statuses` methods that are symbol(s) specific or global for all symbols.

### LINKED ISSUE #427: [Bug]: The option data stream should not have subscribe bar method
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

`OptionDataStream` has a `subscribe_bars` method, but the option stream does not support this.

### Expected Behavior

_No response_

### SDK Version I encountered this issue in

v0.19.1

### Steps To Reproduce

```markdown
-
```

### Filled out the Steps to Reproduce section?

- [X] I have entered valid steps to reproduce my issue or have attached a minimally reproducible case in code that shows my issue happening; and understand that without this my issue will be flagged as invalid and closed after 30 days.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by gnvk**
**Summary of changes:**

- Crypto
  - Add orderbooks
- News: add news stream
- Options
  - Remove bars (Fixes #427) 
- Stocks
  - Add trading statuses (Fixes #309)
  - Add trade corrections
  - Add trade cancel / errors
- Misc.
  - Move the base stream from `alpaca.common.websocket` to `alpaca.data.live.websocket` and renamed it to `DataStream`
  - Move the subscribe / unsubscribe methods to the child classes
  - Fix type annotations
  - Simplify the unsubscribe methods

### CODE REVIEW

File: alpaca/data/models/news.py [hunk 1]
```diff
@@ -17,6 +20,8 @@ class NewsImage(BaseModel):
     size: NewsImageSize
     url: str
 
+    model_config = ConfigDict(protected_namespaces=tuple())
```
Comment (hiohiohio) [inline alpaca/data/models/news.py:23]: could you try to remove this? I have merged a PR for this. Therefore, doc generation process should work without this field.

File: alpaca/data/models/news.py [hunk 2]
```diff
@@ -44,10 +49,20 @@ class News(BaseModel):
     summary: str
     content: str
     url: Optional[str]
-    images: List[NewsImage]
+    images: Optional[List[NewsImage]] = None  # only in historical
```
Comment (hiohiohio) [inline alpaca/data/models/news.py:52]: Could you please docstring as well?

File: alpaca/data/models/news.py [hunk 3]
```diff
@@ -44,10 +49,20 @@ class News(BaseModel):
...
     symbols: List[str]
     source: str
 
+    model_config = ConfigDict(protected_namespaces=tuple())
```
Comment (hiohiohio) [inline alpaca/data/models/news.py:56]: same as above, please try to remove

File: alpaca/data/models/news.py [hunk 4]
```diff
@@ -60,3 +75,5 @@ class NewsSet(BaseModel):
 
     news: List[News]
     next_page_token: Optional[str]
+
+    model_config = ConfigDict(protected_namespaces=tuple())
```
Comment (hiohiohio) [inline alpaca/data/models/news.py:79]: same as above, please try to remove

File: alpaca/data/models/orderbooks.py [hunk 5]
```diff
@@ -31,6 +32,7 @@ class Orderbook(BaseModel):
     timestamp: datetime
     bids: List[OrderbookQuote]
     asks: List[OrderbookQuote]
+    reset: bool = False
```
Comment (hiohiohio) [inline alpaca/data/models/orderbooks.py:35]: Could you please add this to docstring as well?

File: alpaca/data/live/websocket.py [hunk 6]
```diff
@@ -151,79 +167,91 @@ async def _consume(self) -> None:
...
+            symbols = msg.get("symbols")
+            for symbol in symbols:
+                handler = self._handlers["news"].get(symbol)
+                if handler:
+                    await handler(self._cast(msg))
+                    return
+            star_handler = self._handlers["news"].get("*")
+            if star_handler:
+                await star_handler(self._cast(msg))
+                return
```
Comment (hiohiohio) [inline alpaca/data/live/websocket.py:233]: Please consider this implementation. What I thought was:
If we have multiple handler for a different symbols, a single news should be dispatched  into the symbol? Also, if there is no handler for a symbol, the star_handler should be called.

https://github.com/alpacahq/alpaca-py/pull/392/files#diff-f1f76d83a1c39c29072972280610b605f9e44c25ccf734321078c9e5b3f7cc38R235-R249

File: alpaca/data/models/news.py (hunk 1 above)
Comment (gnvk) [reply alpaca/data/models/news.py:23]: After removing this doc generation fails with
```
autodoc: failed to import class 'live.crypto.CryptoDataStream' from module 'alpaca.data'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/gnvk/.pyenv/versions/alpaca-py/lib/python3.11/site-packages/sphinx/ext/autodoc/importer.py", line 143, in import_module
    return importlib.import_module(modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gnvk/.pyenv/versions/3.11.5/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/gnvk/Dev/alpaca/clients/alpaca-py/alpaca/data/live/__init__.py", line 1, in <module>
    from alpaca.data.live.crypto import CryptoDataStream
  File "/Users/gnvk/Dev/alpaca/clients/alpaca-py/alpaca/data/live/crypto.py", line 5, in <module>
    from alpaca.data.live.websocket import DataStream
  File "/Users/gnvk/Dev/alpaca/clients/alpaca-py/alpaca/data/live/websocket.py", line 22, in <module>
    from alpaca.data.models.news import News
  File "/Users/gnvk/Dev/alpaca/clients/alpaca-py/alpaca/data/models/news.py", line 9, in <module>
    class NewsImage(BaseModel):
  File "/Users/gnvk/.pyenv/versions/alpaca-py/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 178, in __new__
    set_model_fields(cls, bases, config_wrapper, types_namespace)
  File "/Users/gnvk/.pyenv/versions/alpaca-py/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 452, in set_model_fields
    fields, class_vars = collect_model_fields(cls, bases, config_wrapper, types_namespace, typevars_map=typevars_map)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/gnvk/.pyenv/versions/alpaca-py/lib/python3.11/site-packages/pydantic/_internal/_fields.py", line 143, in collect_model_fields
    raise NameError(
NameError: Field "model_fields" conflicts with member {} of protected namespace "model_".

make: *** [html] Error 2
```

Comment (gnvk) [reply alpaca/data/models/news.py:23]: Adding `import alpaca.data.models.news  # noqa # pylint: disable=unused-import` to docs/conf.py` fixes this.

File: alpaca/data/live/websocket.py [hunk 7]
```diff
@@ -2,22 +2,31 @@
...
+from alpaca.data.models import (
+    Bar,
+    Orderbook,
+    Quote,
+    Trade,
+    TradeCancel,
+    TradeCorrection,
+    TradingStatus,
+)
+from alpaca.data.models.news import News
```
Comment (hiohiohio) [inline alpaca/data/live/websocket.py:22]: this News also can be a part of the import list above as we import news in models/__init__.py?

**Review by hiohiohio — APPROVED** [review]
LGTM!

File: alpaca/data/live/stock.py [hunk 8]
```diff
@@ -51,3 +53,147 @@ def __init__(
...
+        Args:
+            handler (Callable[[Union[TradeCorrection, Dict]]): The coroutine callback
+                function to handle the incoming data.
+        """
+        self._handlers["corrections"] = {"*": handler}
+
+    def register_trade_cancels(
+        self, handler: Callable[[Union[TradeCancel, Dict]], Awaitable[None]]
+    ) -> None:
+        """Register a trade cancel handler. You can only subscribe to trade corrections by
```
Comment (hiohiohio) [inline alpaca/data/live/stock.py:144]: [minor]
```suggestion
        """Register a trade cancel handler. You can only subscribe to trade cancels by
```

**Pull request merged**